lib/cib/cib_utils.c

```diff
@@ -77,6 +78,116 @@ cib_diff_version_details(xmlNode * diff, int *admin_epoch, int *epoch, int *upda
     return TRUE;
 }
 
+/*!
+ * \internal
+ * \brief Check whether a given CIB element was modified in a v1 update diff
+ *
+ * \param[in] patchset  XML patchset from CIB update
+ * \param[in] element   XML tag of CIB element to check
+ *
+ * \return \c true if \p element was modified in the CIB update, or \c false
+ *         otherwise
+ */
+static bool
+element_in_patchset_v1(const xmlNode *patchset, const char *element)
+{
+    char *xpath = crm_strdup_printf("//" F_CIB_UPDATE_RESULT
+                                    "//" XML_TAG_DIFF_ADDED "//%s",
+                                    element);
+    xmlXPathObject *xpath_obj = xpath_search(patchset, xpath);
+
+    free(xpath);
+
+    if (xpath_obj == NULL) {
+        return false;
+    }
+    freeXpathObject(xpath_obj);
+    return true;
+}
+
+/*!
+ * \internal
+ * \brief Check whether a given CIB element was modified in a v2 update diff
+ *
+ * \param[in] patchset  XML patchset from CIB update
+ * \param[in] element   XML tag of CIB element to check
+ *
+ * \return \c true if \p element was modified in the CIB update, or \c false
+ *         otherwise
+ *
+ * \note Any CIB element supported by \c pcmk__cib_abs_xpath_for() is a
+ *       supported value for \p element.
+ */
+static bool
+element_in_patchset_v2(const xmlNode *patchset, const char *element)
+{
+    const char *xpath_element = pcmk__cib_abs_xpath_for(element);
+    const char *xpath_parent = pcmk_cib_parent_name_for(element);
+
+    for (const xmlNode *change = first_named_child(patchset, XML_DIFF_CHANGE);
+         change != NULL; change = crm_next_same_xml(change)) {
+
+        const char *xpath = crm_element_value(change, XML_DIFF_PATH);
+
+        if (pcmk__starts_with(xpath, xpath_element)) {
+            // Change to an existing element
+            return true;
+        }
+
+        if (pcmk__str_eq(xpath, xpath_parent, pcmk__str_none)
+            && pcmk__xe_is(pcmk__xml_first_child(change), element)) {
+
+            // Newly added element in a create operation
+            return true;
+        }
+    }
+    return false;
+}
+
+/*!
+ * \internal
+ * \brief Check whether a given CIB element was modified in a CIB update
+ *
+ * \param[in] msg      CIB update result (from a diff notification)
+ * \param[in] element  XML tag of CIB element to check
+ *
+ * \return \c true if \p element was modified in the CIB update, or \c false
+ *         otherwise
+ *
+ * \note Any CIB element supported by \c pcmk__abs_cib_xpath_for() is a
+ *       supported value for \p element.
+ */
+bool
+cib__element_in_patchset(const xmlNode *msg, const char *element)
```

Comment:
I haven't decided what approach I'm going to use when we want to do things soon like "check whether the status section OTHER THAN transient attributes changed."

I might convert this to `cib__xpath_in_patchset()` (or create a new such function) that does regex matching. It may or may not be able to support XPaths with multiple terms (e.g., `/path1|/path2`).